presences/src/main/resources/public/ts/controllers/alerts.ts

```diff
@@ -93,166 +97,160 @@ export const alertsController = ng.controller('AlertsController', ['$scope', '$r
+        } else {
+            if (this.params.type) {
+                this.initFilter(false);
+                this.filter.types[this.params.type] = true;
             }
-        };
-
-        vm.someSelectedAlert = function () {
-            return vm.listAlert && vm.listAlert.filter(alert => alert.selected).length > 0;
-        };
+            await this.viescolaireService.getSchoolYearDates(window.structure.id)
+                .then((schoolYears: ISchoolYearPeriod) => this.filter.startDate = moment(schoolYears.start_date))
+                .catch(e => console.error(e));
+            await this.getStudentAlert();
+        }
+    };
 
-        vm.selectStudent = async function (model, student) {
-            if (_.findWhere(vm.filter.selected.students, {id: student.id})) {
-                return;
-            }
-            vm.filter.selected.students.push(student);
-            vm.filter.student = '';
-            vm.filter.students = undefined;
-            await vm.getStudentAlert(extractSelectedStudentIds(), extractSelectedGroupsName());
-            $scope.safeApply();
-        };
+    initFilter(value: boolean): void {
+        this.filters.forEach(filter => this.filter.types[filter] = value);
+    };
 
-        vm.searchClass = async function (value) {
-            const structureId = window.structure.id;
-            try {
-                vm.filter.classes = await GroupService.search(structureId, value);
-                vm.filter.classes.map((obj) => obj.toString = () => obj.name);
-                $scope.safeApply();
-            } catch (err) {
-                vm.filter.classes = [];
-                throw err;
+    async getStudentAlert(students?: string[], classes?: string[]): Promise<void> {
+        this.params.loading = true;
+        safeApply(this.$scope);
+        this.alertType = [];
+        Object.keys(this.filter.types).forEach(key => {
+            if (this.filter.types[key]) this.alertType.push(key);
+        });
+
+        try {
+            if (this.alertType.length > 0) {
+                const start_at: string = moment(this.filter.startDate).format(DateUtils.FORMAT["YEAR-MONTH-DAY"]);
+                const end_at: string = moment(this.filter.endDate).format(DateUtils.FORMAT["YEAR-MONTH-DAY"]);
+                const studentsAlerts: any = await this.alertService.getStudentsAlerts(window.structure.id, this.alertType, students, classes, start_at, end_at);
+                this.listAlert = studentsAlerts;
+            } else {
+                this.listAlert = [];
             }
+        } catch (e) {
+            toasts.warning('presences.error.get.alert');
+            throw e;
+        }
+        this.params.loading = false;
+        safeApply(this.$scope);
+    }
+    async searchStudent(value: string): Promise<void> {
+        const structureId = window.structure.id;
+        try {
+            that.filter.students = await that.searchService.searchUser(structureId, value, 'Student');
```

Comment:
utilisation de that car cette fonction est appeler par un autre conposant. Ce qui fait perdre l'instance du this